### ISSUE TITLE
sorry，could you explain why GraspNet doesn't work for industrial parts? 

### ISSUE THREAD

**Issue by biosbeta** [sunhan1997/IndusGrasp#2]
From the images  include grasping poses，it seems that some of the grasping poses provided by graspnet are reasonable and feasible.So I didn't understand why GraspNet doesn't work for industrial parts,could you explain that?

**Comment by sunhan1997** [comment]
Does "image" refer to the image with the blue border in the README? GraspNet works well when the scene contains objects of large size, and the point cloud features of these objects are obvious enough. However, when there are only metal objects in the scene, the point cloud features of these objects are not obvious enough (including Reaslsense and 3D industrial cameras), and GraspNet does not work (red border image in README). The reasons are as follows: 1. Metal objects are small and reflective, and the quality of the point cloud is poor, and GraspNet is very dependent on the point cloud. 2. GraspNet is not trained for bin-picking scenarios. In addition, I have tested the ROBI dataset with GraspNet and it still does not work.

**Comment by sunhan1997** [comment]
Does "image" refer to the image with the blue border in the README? GraspNet works well when the scene contains objects of large size, and the point cloud features of these objects are obvious enough. However, when there are only metal objects in the scene, the point cloud features of these objects are not obvious enough (including Reaslsense and 3D industrial cameras), and GraspNet does not work (red border image in README). The reasons are as follows: 1. Metal objects are small and reflective, and the quality of the point cloud is poor, and GraspNet is very dependent on the point cloud. 2. GraspNet is not trained for bin-picking scenarios. In addition, I have tested the ROBI dataset with GraspNet and it still does not work.




| |
孙晗
|
|
***@***.***
|
---- Replied Message ----
| From | ***@***.***> |
| Date | 5/21/2024 22:29 |
| To | ***@***.***> |
| Cc | ***@***.***> |
| Subject | [sunhan1997/IndusGrasp] sorry，could you explain why GraspNet doesn't work for industrial parts? (Issue #2) |

From the images include grasping poses，it seems that some of the grasping poses provided by graspnet are reasonable and feasible.So I didn't understand why GraspNet doesn't work for industrial parts,could you explain that?

**Comment by biosbeta** [comment]
谢谢你的回复，不过代码好像跑不通，跑test.py 报错没有utils.kps_heatmap和utils.eval_grasp_util

**Comment by sunhan1997** [comment]
不好意思，我记得以及上传了，可能没有上传成功，我重新上传一下







| |
孙晗
|
|
***@***.***
|
---- Replied Message ----
| From | ***@***.***> |
| Date | 5/23/2024 21:06 |
| To | ***@***.***> |
| Cc | ***@***.***>,
***@***.***> |
| Subject | Re: [sunhan1997/IndusGrasp] sorry，could you explain why GraspNet doesn't work for industrial parts? (Issue #2) |

谢谢你的回复，不过代码好像跑不通，跑test.py 报错没有utils.kps_heatmap和utils.eval_grasp_util

**Comment by sunhan1997** [comment]
你好，我更新了github上的代码，但我现在网络不太好，还有两个文件需要替换，附件发给你。另外物体检测的权重在这里“链接：https://pan.baidu.com/s/1xCyfScKi782zLlB5I_3_kg 
提取码：3vc6”，可以放在example文件夹中。运行test.py即可。如果依旧有问题，欢迎随时提问，会尽快回复的。







| |
孙晗
|
|
***@***.***
|
---- Replied Message ----
| From | ***@***.***> |
| Date | 5/23/2024 21:06 |
| To | ***@***.***> |
| Cc | ***@***.***>,
***@***.***> |
| Subject | Re: [sunhan1997/IndusGrasp] sorry，could you explain why GraspNet doesn't work for industrial parts? (Issue #2) |

谢谢你的回复，不过代码好像跑不通，跑test.py 报错没有utils.kps_heatmap和utils.eval_grasp_util

**Comment by biosbeta** [comment]
issue是看不到附件的，那两个需要替换的文件，麻烦你传个百度网盘，发个地址吧。

**Comment by sunhan1997** [comment]
百度云盘：链接：https://pan.baidu.com/s/1n74lnKrEyxmJY1VaErFVvw 
提取码：fk3c


| |
孙晗
|
|
***@***.***
|
---- Replied Message ----
| From | ***@***.***> |
| Date | 5/24/2024 16:33 |
| To | ***@***.***> |
| Cc | ***@***.***>,
***@***.***> |
| Subject | Re: [sunhan1997/IndusGrasp] sorry，could you explain why GraspNet doesn't work for industrial parts? (Issue #2) |

issue是看不到附件的，那两个需要替换的文件，麻烦你传个百度网盘，发个地址吧。

**Comment by biosbeta** [comment]
![2024-05-27 09-45-11 的屏幕截图](https://github.com/sunhan1997/IndusGrasp/assets/120536733/014f7e8d-3d15-419b-9eff-b8c8736a01d2)
我得到了很奇怪的结果，抓取关键点（绿色、蓝色）和抓取点（红色）并不在一条直线上，同时代表抓取角度的黄线也没有显示

**Comment by sunhan1997** [comment]
看起来是物体检测部分失效了，导致整个图像输入到了网络中。实际上应该是检测出一个物体，并将一个物体的patch输入网络。另外代码里面有个angle_grasp的参数表示的是抓取角度，这段代码中没有画图的部分。如果方便的话，可以加我微信（sh15938469663）







| |
孙晗
|
|
***@***.***
|
---- Replied Message ----
| From | ***@***.***> |
| Date | 5/27/2024 09:40 |
| To | ***@***.***> |
| Cc | ***@***.***>,
***@***.***> |
| Subject | Re: [sunhan1997/IndusGrasp] sorry，could you explain why GraspNet doesn't work for industrial parts? (Issue #2) |

2024-05-27.09-35-10.png (view on web)
我得到了很奇怪的结果，抓取关键点（绿色、蓝色）和抓取点（红色）并不在一条直线上，同时代表抓取角度的黄线也没有显示

**Comment by Henrymarine** [comment]
![QQ图片20240531145729](https://github.com/sunhan1997/IndusGrasp/assets/166482264/9aaaddc6-d3a1-4ef2-924c-c9fc14861f73)
不好意思打扰一下，我在运行train文件的时候报错了，在定义的resnet18也没有发现scheduler，请问这是因为什么，能够帮忙解答一下吗

**Comment by sunhan1997** [comment]
> ![QQ图片20240531145729](https://private-user-images.githubusercontent.com/166482264/335474806-9aaaddc6-d3a1-4ef2-924c-c9fc14861f73.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTcxMzk5NTcsIm5iZiI6MTcxNzEzOTY1NywicGF0aCI6Ii8xNjY0ODIyNjQvMzM1NDc0ODA2LTlhYWFkZGM2LWQzYTEtNGVmMi05MjRjLWM5ZmMxNDg2MWY3My5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTMxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUzMVQwNzE0MTdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03ZDM4Y2Y0MzEzZDViMjhlMDcwNDg3YmU1NzYxMjExOGY2NTFkYzhlZjAwODJkNzFiODQzYzE4ZGY2M2NjYmVmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.o_bREDhWnuRLCIvrmTjUHLHM4Ee3qBM9HdE3tJJDXR4)不好意思打扰一下，我在运行train文件的时候报错了，在定义的resnet18也没有发现scheduler，请问这是因为什么，能够帮忙解答一下吗

  可能torch版本有影响，你把  print("learning_rate: ", model.scheduler.get_last_lr())注释掉试试

**Comment by Henrymarine** [comment]
如果注释了就是可以的


---- Replied Message ----
| From | ***@***.***> |
| Date | 5/31/2024 15:16 |
| To | ***@***.***> |
| Cc | ***@***.***> ,
***@***.***> |
| Subject | Re: [sunhan1997/IndusGrasp] sorry，could you explain why GraspNet doesn't work for industrial parts? (Issue #2) |

不好意思打扰一下，我在运行train文件的时候报错了，在定义的resnet18也没有发现scheduler，请问这是因为什么，能够帮忙解答一下吗

可能torch版本有影响，你把 print("learning_rate: ", model.scheduler.get_last_lr())注释掉试试

**Comment by Henrymarine** [comment]
> 谢谢你的回复，不过代码好像跑不通，跑 test.py 报错没有utils.kps_heatmap和utils.eval_grasp_util

![image](https://github.com/sunhan1997/IndusGrasp/assets/166482264/3f905c26-3edf-4bbc-876f-4bb37e419471)
请问你在运行test的时候有没有遇到这个问题，如果遇到了您是怎么解决的

**Comment by biosbeta** [comment]
没有

> > 谢谢你的回复，不过代码好像跑不通，跑 test.py 报错没有utils.kps_heatmap和utils.eval_grasp_util
> 
> ![image](https://private-user-images.githubusercontent.com/166482264/335845933-3f905c26-3edf-4bbc-876f-4bb37e419471.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTczNzQ1MzgsIm5iZiI6MTcxNzM3NDIzOCwicGF0aCI6Ii8xNjY0ODIyNjQvMzM1ODQ1OTMzLTNmOTA1YzI2LTNlZGYtNGJiYy04NzZmLTRiYjM3ZTQxOTQ3MS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYwM1QwMDIzNThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xNzEyMzc1ODBiZTVjMThhODI4NjI3MjdhYmU0MTQxMDUyZWU0YzdiOWNhZjVjZDY4MzliNjk4YTVmNWFjNDliJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.3k5CJN1ULxxxGeEjRKRHs70jF1OS0G_FO9dm6yHbMZw) 请问你在运行test的时候有没有遇到这个问题，如果遇到了您是怎么解决的

没遇到过。不过，看报错像是没获取到all-score，可能是推理过程中模型没检测出抓姿？

**Comment by sunhan1997** [comment]
> > 谢谢你的回复，不过代码好像跑不通，跑 test.py 报错没有utils.kps_heatmap和utils.eval_grasp_util
> 
> ![image](https://private-user-images.githubusercontent.com/166482264/335845933-3f905c26-3edf-4bbc-876f-4bb37e419471.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc0MTQ3MzAsIm5iZiI6MTcxNzQxNDQzMCwicGF0aCI6Ii8xNjY0ODIyNjQvMzM1ODQ1OTMzLTNmOTA1YzI2LTNlZGYtNGJiYy04NzZmLTRiYjM3ZTQxOTQ3MS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYwM1QxMTMzNTBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00MDVmZGIyMzc2ZjQ4Yjc2YWU5ZTgzZjcwZTZhNzExMTdlODBlNWFjNGJmNTljNjNjYjAxODJjMDRjYWExMDRjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.bawiUjpEWxx-63mTpqcctec39MfdUUpc-Zj0FfYXSN4) 请问你在运行test的时候有没有遇到这个问题，如果遇到了您是怎么解决的

这部分应该是物体检测的问题，你先看看fastrcnn这部分检测是否正常

**Comment by Henrymarine** [comment]
好的好的谢谢您了，我再仔细看看
---- Replied Message ----
| From | ***@***.***> |
| Date | 6/3/2024 19:35 |
| To | ***@***.***> |
| Cc | ***@***.***> ,
***@***.***> |
| Subject | Re: [sunhan1997/IndusGrasp] sorry，could you explain why GraspNet doesn't work for industrial parts? (Issue #2) |

谢谢你的回复，不过代码好像跑不通，跑 test.py 报错没有utils.kps_heatmap和utils.eval_grasp_util

请问你在运行test的时候有没有遇到这个问题，如果遇到了您是怎么解决的

这部分应该是物体检测的问题，你先看看fastrcnn这部分检测是否正常
